# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xf86-video-s3virge/hotfix.patch
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

From 6c66938ca97c2a24f96856c2e32010af052e9a6e Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Sun, 30 Jan 2022 12:20:22 -0800
Subject: [PATCH] Move [HV]Total checks into S3VValidMode

Needed to build against Xorg 1.20 and later due to xorg/xserver@5a945f54
Based on ajax's commit xf86-video-rendition@70dd6c22

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 src/s3v_driver.c | 21 +++++++++------------
 1 file changed, 9 insertions(+), 12 deletions(-)

diff --git a/src/s3v_driver.c b/src/s3v_driver.c
index b39f410..5a8718a 100644
--- a/src/s3v_driver.c
+++ b/src/s3v_driver.c
@@ -1201,18 +1201,6 @@ S3VPreInit(ScrnInfoPtr pScrn, int flags)
    vga256InfoRec.directMode = XF86DGADirectPresent;
 #endif
 
-    /*
-     * xf86ValidateModes will check that the mode HTotal and VTotal values
-     * don't exceed the chipset's limit if pScrn->maxHValue and
-     * pScrn->maxVValue are set.  
-     */
-
-    /* todo -  The virge limit is 2048 vertical & horizontal */
-    /* pixels, not clock register settings. */
-			 	/* true for all ViRGE? */
-  pScrn->maxHValue = 2048;
-  pScrn->maxVValue = 2048;
-
     				/* Lower depths default to config file */
   pScrn->virtualX = pScrn->display->virtualX;
 				/* Adjust the virtualX to meet ViRGE hardware */
@@ -2564,6 +2552,15 @@ S3VValidMode(SCRN_ARG_TYPE arg, DisplayModePtr mode, Bool verbose, int flags)
     if ((pScrn->bitsPerPixel + 7)/8 * mode->HDisplay > 4095)
 	return MODE_VIRTUAL_X;
 
+    /* todo -  The virge limit is 2048 vertical & horizontal */
+    /* pixels, not clock register settings. */
+				/* true for all ViRGE? */
+    if (mode->HTotal > 2048)
+        return MODE_BAD_HVALUE;
+
+    if (mode->VTotal > 2048)
+        return MODE_BAD_VVALUE;
+
     return MODE_OK;
 }
 
-- 
GitLab

